fairlearn/reductions/_grid_search/grid_search.py

```diff
@@ -165,12 +167,20 @@ def fit(self, X, y, **kwargs):
             else:
                 y_reduction = y_train
 
-            current_estimator = copy.deepcopy(self.estimator)
-            logger.debug("Calling underlying estimator")
-            oracle_call_start_time = time()
-            current_estimator.fit(X, y_reduction, sample_weight=weights)
-            oracle_call_execution_time = time() - oracle_call_start_time
-            logger.debug("Call to underlying estimator complete")
+            y_reduction_unique = np.unique(y_reduction)
+            if len(y_reduction_unique) == 1:
+                logger.debug("y_reduction had single value. Using DummyClassifier")
+                current_estimator = DummyClassifier(strategy='constant',
+                                                    constant=y_reduction_unique[0])
+                current_estimator.fit(X, y_reduction, sample_weight=weights)
```

Comment:
Similar to `ExponentiatedGradient`, can we pull out `fit` and time tracking out of `if`?